exist-core/src/test/xquery/util/get-fragment-between.xqm

```diff
@@ -0,0 +1,85 @@
+xquery version "3.1";
+
+module namespace gfb = "http://exist-db.org/test/util/get-fragment-between";
+
+declare namespace test = "http://exist-db.org/xquery/xqsuite";
+declare namespace tei = "http://www.tei-c.org/ns/1.0";
+
+declare variable $gfb:DOC1 :=
+    <TEI xmlns="http://www.tei-c.org/ns/1.0">
+        <text>
+            <front>
+                <pb facs="1.jpg"/>
+                <p>Aus dem Leben einer Kartoffel.</p>
+                <pb facs="2.jpg"/>
+            </front>
+            <body>
+                <pb facs="3.jpg"/>
+                <p>Hubertus Knoll spazierte über das <pb facs="4.jpg"/> Feld.</p>
+                <pb facs="5.jpg"/>
+            </body>
+        </text>
+    </TEI>;
+
+declare variable $gfb:DOC2 :=
+    <root xmlns="http://exist-db.org/xquery/xqsuite">
+        <x/>
+        <y/>
+        <z/>
+    </root>;
+
+declare
+    %test:setUp
+function gfb:setup() {
+    xmldb:create-collection("/db", "test-gfb"),
+    xmldb:store("/db/test-gfb", "doc1.xml", $gfb:DOC1),
+    xmldb:store("/db/test-gfb", "doc2.xml", $gfb:DOC2)
+};
+
+declare
+    %test:tearDown
+function gfb:teardown() {
+    xmldb:remove("/db/test-gfb")
+};
+
+declare
+    %test:assertEquals('<pb facs="1.jpg"></pb><p>Aus dem Leben einer Kartoffel.</p>')
```

Comment:
The fact that the function's 4th parameter is `true()` means the resulting fragment should preserve the root namespace:

> The fourth argument, `$display-root-namespace`, is a boolean value for displaying the root node namespace. If it is set to `true()` the attribute `xmlns` in the root node of the result sequence is determined explicitly from the `$beginning-node`.

```suggestion
    %test:assertEquals('<pb xmlns="http://www.tei-c.org/ns/1.0" facs="1.jpg"></pb><p>Aus dem Leben einer Kartoffel.</p>')
```

Reply:
I think the 4th parameter only has an effect when the 3rd parameter is `true`. I have committed two further tests to try and illustrate this.

Unfortunately it is very hard to know what the 'correct' behaviour of `util:get-fragment-between` is exactly mean to be as there is no documentation :-/

Reply:
Good point about the thin documentation for this function (explaining why I wasn’t sure whether the second node was preserved or discarded). 

But is it necessary to limit namespace preservation to when a root node is added? Surely users would need a way to preserve the namespace of the resulting nodes. 

My suggestion also neglected to include the namespace on the 2nd node.

Reply:
> But is it necessary to limit namespace preservation to when a root node is added? Surely users would need a way to preserve the namespace of the resulting nodes.

No... but my goal was to fix the regression, and just get it working as it was in eXist-db 4.3.1.